staging/src/k8s.io/kubectl/pkg/cmd/annotate/annotate.go

```diff
@@ -137,132 +154,146 @@ func NewCmdAnnotate(parent string, f cmdutil.Factory, ioStreams genericclioption
 		Example:               annotateExample,
 		ValidArgsFunction:     completion.ResourceTypeAndNameCompletionFunc(f),
 		Run: func(cmd *cobra.Command, args []string) {
-			cmdutil.CheckErr(o.Complete(f, cmd, args))
-			cmdutil.CheckErr(o.Validate())
-			cmdutil.CheckErr(o.RunAnnotate())
+			o, err := flags.ToOptions(f, cmd, args)
+			cmdutil.CheckErr(err)
+			cmdutil.CheckErr(o.RunAnnotate(flags))
 		},
 	}
 
-	// bind flag structs
-	o.RecordFlags.AddFlags(cmd)
-	o.PrintFlags.AddFlags(cmd)
+	flags.AddFlags(cmd, streams)
+
+	return cmd
+}
+
+// AddFlags registers flags for a cli.
+func (flags *AnnotateFlags) AddFlags(cmd *cobra.Command, ioStreams genericclioptions.IOStreams) {
+	o := NewAnnotateOptions(ioStreams)
```

Comment:
I think, all the flags might have a corresponding object in flags and this should not be in here. Because it looks to me that this is breaking the loose coupling and the simplicity of flow(get flags, get options from flags and options.run without flags)

Reply:
@ardaguclu, of course, what you are saying makes totally sense. I tried to update accordingly :)